examples/platform/silabs/MatterShell.cpp

```diff
@@ -112,6 +117,154 @@ void cmdSilabsInit()
 
 #endif // SL_CATALOG_CLI_PRESENT
 
+#if MATTER_TRACING_ENABLED
+using TimeTraceOperation = Tracing::Silabs::TimeTraceOperation;
+
+TimeTraceOperation StringToTimeTraceOperation(const char * str)
+{
+    if (strcmp(str, "Spake2p") == 0)
+    {
+        return TimeTraceOperation::kSpake2p;
+    }
+    else if (strcmp(str, "Pake1") == 0)
+    {
+        return TimeTraceOperation::kPake1;
+    }
+    else if (strcmp(str, "Pake2") == 0)
+    {
+        return TimeTraceOperation::kPake2;
+    }
+    else if (strcmp(str, "Pake3") == 0)
+    {
+        return TimeTraceOperation::kPake3;
+    }
+    else if (strcmp(str, "OperationalCredentials") == 0)
+    {
+        return TimeTraceOperation::kOperationalCredentials;
+    }
+    else if (strcmp(str, "AttestationVerification") == 0)
+    {
+        return TimeTraceOperation::kAttestationVerification;
+    }
+    else if (strcmp(str, "CSR") == 0)
+    {
+        return TimeTraceOperation::kCSR;
+    }
+    else if (strcmp(str, "NOC") == 0)
+    {
+        return TimeTraceOperation::kNOC;
+    }
+    else if (strcmp(str, "TransportLayer") == 0)
+    {
+        return TimeTraceOperation::kTransportLayer;
+    }
+    else if (strcmp(str, "TransportSetup") == 0)
+    {
+        return TimeTraceOperation::kTransportSetup;
+    }
+    else if (strcmp(str, "FindOperational") == 0)
+    {
+        return TimeTraceOperation::kFindOperational;
+    }
+    else if (strcmp(str, "CaseSession") == 0)
+    {
+        return TimeTraceOperation::kCaseSession;
+    }
+    else if (strcmp(str, "Sigma1") == 0)
+    {
+        return TimeTraceOperation::kSigma1;
+    }
+    else if (strcmp(str, "Sigma2") == 0)
+    {
+        return TimeTraceOperation::kSigma2;
+    }
+    else if (strcmp(str, "Sigma3") == 0)
+    {
+        return TimeTraceOperation::kSigma3;
+    }
+    else if (strcmp(str, "OTA") == 0)
+    {
+        return TimeTraceOperation::kOTA;
+    }
+    else if (strcmp(str, "ImageUpload") == 0)
+    {
+        return TimeTraceOperation::kImageUpload;
+    }
+    else if (strcmp(str, "ImageVerification") == 0)
+    {
+        return TimeTraceOperation::kImageVerification;
+    }
+    else if (strcmp(str, "AppApplyTime") == 0)
+    {
+        return TimeTraceOperation::kAppApplyTime;
+    }
+    else if (strcmp(str, "Bootup") == 0)
+    {
+        return TimeTraceOperation::kBootup;
+    }
+    else if (strcmp(str, "SilabsInit") == 0)
+    {
+        return TimeTraceOperation::kSilabsInit;
+    }
+    else if (strcmp(str, "MatterInit") == 0)
+    {
+        return TimeTraceOperation::kMatterInit;
+    }
+    else if (strcmp(str, "BufferFull") == 0)
+    {
+        return TimeTraceOperation::kBufferFull;
+    }
+    else
+    {
+        return TimeTraceOperation::kNumTraces;
+    }
+}
+
+CHIP_ERROR CmdTracingDispatch(int argc, char ** argv)
```

Comment:
You can have subcommands managed directly be the shell structures init. This avoid having huge if else statements to manages multiple commands.

Reply:
Probably, I was just trying to imitate the style above.

Also:
> shell structures init.

Are you proposing we create a child class to the Shell class?

Reply:
He means that you can have SubCommands handler. That was done for LightSwitch and Door lock app shell commands.

Personally, I'm fine with either or. Check what Math means if you prefer it